pylint/config.py

```diff
@@ -494,52 +358,82 @@ def add_option_group(self, group_name, _, options, provider):
         if group_name in self._mygroups:
             group = self._mygroups[group_name]
         else:
-            group = optparse.OptionGroup(self.cmdline_parser,
-                                         title=group_name.capitalize())
-            self.cmdline_parser.add_option_group(group)
-            group.level = provider.level
+            group = self.cmdline_parser._parser.add_argument_group(
+                group_name.capitalize(), level=provider.level,
+            )
             self._mygroups[group_name] = group
             # add section to the config file
-            if group_name != "DEFAULT" and \
-                    group_name not in self.cfgfile_parser._sections:
-                self.cfgfile_parser.add_section(group_name)
+            if group_name != "DEFAULT":
+                try:
+                    self.cfgfile_parser._parser.add_section(group_name)
+                except configparser.DuplicateSectionError:
+                    pass
+
         # add provider's specific options
         for opt, optdict in options:
             self.add_optik_option(provider, group, opt, optdict)
 
     def add_optik_option(self, provider, optikcontainer, opt, optdict):
         args, optdict = self.optik_option(provider, opt, optdict)
-        option = optikcontainer.add_option(*args, **optdict)
+        if hasattr(optikcontainer, '_parser'):
+            optikcontainer = optikcontainer._parser
```

Comment:
I hope we can get rid of this ``optik`` names later on.

Reply:
Me too. I'll be phasing it out as more and more of this config logic disappears.